File: src/core/payment-request/dto/create-payment-request.dto.ts
import { IsInt, Max, Min } from 'class-validator';

export class CreatePaymentRequestDto {
  @IsInt({ message: 'Digite um valor válido' })
  @Min(10, { message: 'Digite um valor de pelo menos dez centavos' })
  @Max(1000000_00, {
    message: 'Digite um valor menor que um milhão de reais',
  })
  public value: number;

  declare public userId: string;
}

File: src/main/java/dev/joaopdias/auronix/core/paymentrequest/dto/CreatePaymentRequestDto.java
package dev.joaopdias.auronix.core.paymentrequest.dto;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;

public record CreatePaymentRequestDto(
    @Min(value = 10, message = "Digite um valor de pelo menos dez centavos")
    @Max(value = 1000000_00, message = "Digite um valor menor que um milhão de reais")
    long value
) {
}
